data-prepper-plugins/opensearch/src/main/java/com/amazon/dataprepper/plugins/sink/opensearch/index/IndexConfiguration.java

```diff
@@ -93,6 +87,31 @@ private IndexConfiguration(final Builder builder) {
         this.ismPolicyFile = builder.ismPolicyFile;
     }
 
+    private void determineIndexType(Builder builder) {
+        if(builder.indexType != null) {
+            Optional<IndexType> mappedIndexType = IndexType.get(builder.indexType);
+            if (mappedIndexType.isPresent()) {
```

Comment:
Optional provides a nice way to clean this up.

```
indexType = mappedIndexType.orElseThrow(() -> new IllegalArgumentException(...))
```

Reply:
Sounds great!
changed to 
            mappedIndexType.ifPresent(type -> {indexType = type;});
            mappedIndexType.orElseThrow(() -> new IllegalArgumentException("Value of the parameter, index_type, must be from the list: "
                    + IndexType.getIndexTypeValues()));